lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -310,9 +329,15 @@ struct ModuleRebuildInfo {
     }
     // We may have found multiple failing modules, that failed for different
     // reasons. Emit a note for each of them.
-    for (auto &mod : outOfDateModules) {
-      diags.diagnose(loc, diag::out_of_date_module_here,
-                         (unsigned)mod.kind, mod.path);
+    for (auto &mod : candidateModules) {
+      // If a the compiled module was ignored, diagnose the reason.
+      if (mod.reasonIgnored != ReasonIgnored::NotIgnored) {
+        diags.diagnose(loc, diag::compiled_module_ignored_reason, mod.path,
+                       (unsigned)mod.reasonIgnored);
+      } else {
+        diags.diagnose(loc, diag::out_of_date_module_here, (unsigned)mod.kind,
+                       mod.path);
+      }
```

Comment:
Is there something limiting when this diagnostics is shown? I'd expect it to be behind the same flag (and build farm env var) as the rebuild remarks.

Reply:
Yes, this is just an addition to an existing function that already handled printing other kinds of diagnostics for explaining why a module was ignored. There is existing logic elsewhere that dictates whether these diagnostics are emitted at all.